### ISSUE TITLE
TURNS (TURN with TLS/DTLS) socket buffer operation error

### ISSUE THREAD

**Issue by 532910** [matrix-org/synapse#5954]
I've installed and configured coturn daemon for tls and plaintext listeners.
It work fine, while connection is plaintext: ```turn_uris: [ "turn:domain.tld:3478?transport=udp", "turn:domain.tld:3478?transport=tcp" ]```

```testssl.sh domain.tld:5349``` and ```testssl.sh domain.tld:3478``` also shows no errors.

But when I use
```turn_uris: [ "turns:domain.tld:5349?transport=udp", "turns:domain.tld:5349?transport=tcp" ]```
or
```turn_uris: [ "turns:domain.tld:3478?transport=udp", "turns:domain.tld:3478?transport=tcp" ]```

turnserver reports an error:  ```session closed ... reason: TLS/TCP socket buffer operation error (callback)```

**Comment by richvdh** [comment]
that sounds more like a client bug than a synapse one? `turn_uris` is used solely to populate the response to the `/voip/turnServer` endpoint.

**Comment by 532910** [comment]
Agree. Should I open new three separates issues for riot-android, riot-web, and riot-ios?

**Comment by richvdh** [comment]
if it's affecting all those clients, I guess so.

**Comment by jryans** [comment]
If all clients are affected, I would maybe start with riot-web and mention it affects the others as well, rather than fragmenting the conversation across 3 issues. (For the moment, it sounds like a TURN configuration issue or coturn problem, but it's hard to be sure.)
